code/game/machinery/smartglass.dm

```diff
@@ -0,0 +1,131 @@
+/////////////////////////////////////////
+// Machine that makes glass 'smart'
+// Held within a pane of glass
+// used for linking to buttons 'n shit
+// made when a floor light tile is attached to a window
+/////////////////////////////////////////
+
+/obj/machinery/smartglass_electronics
+	name = "smartglass electronics"
+	desc = "This should be inside a pane of smart glass. How are you seeing this?"
+	icon = 'icons/obj/airlock_machines.dmi'
+	icon_state = "access_control_standby"
+	anchored = 0
+	density = 0
+	use_power = 1
+	idle_power_usage = 1
+	active_power_usage = 50
+	power_channel = ENVIRON
+	machine_flags = MULTITOOL_MENU
+	
+	//Smartglass vars
+	var/smart_transparency = 0
+	var/obj/structure/window/GLASS //Ref to the window we're in
```

Comment:
Don't use ALLCAPS variable names, that's for constants.

Reply:
roger